query-node/substrate-query-framework/index-builder/src/db/db.ts

```diff
@@ -0,0 +1,57 @@
+import * as shortid from 'shortid';
+import { Connection, EntityManager } from 'typeorm';
+import { EventHistory } from '.';
+import { QueryEvent } from '..';
+
+export default class DB {
+  private readonly _connection: Connection;
+
+  constructor(connection: Connection) {
+    this._connection = connection;
+  }
+
+  /**
+   * Return the last processed event. If does not exists create a new one
+   */
+  async getLastProcessedEvent(event: QueryEvent): Promise<EventHistory> {
+    let eventHistory = await this._connection.getRepository(EventHistory).findOne();
+    if (!eventHistory) {
+      eventHistory = new EventHistory();
+    }
+    eventHistory.index = event.index;
+    eventHistory.eventName = event.event_method;
+    eventHistory.blockNumber = event.block_number;
+    return eventHistory;
+  }
+
+  /**
+   * Fixes compatibility between typeorm and warthog models
+   * @param entity
+   */
+  fillRequiredWarthogFields(entity: any) {
```

Comment:
What is going on here? I did not understand this.

Also, this is not really a proper object method, this seems like a static method at best, if not a standalone function, but its hard to say given that its not clear to me what is going on.

Reply:
Database models are generated with warthog CLI. Every generated model extends warthog's `BaseModel` which has default these. And these fields are required (btw I also don't like it).

I agree it could be a static or better a standalone utility function.